### ISSUE TITLE
Multiple graphql query variables

### ISSUE THREAD

**Issue by jbolda** [gatsbyjs/gatsby#2986]
I presume you can have more than one query variable? It works in graphiql and appears to actually return the proper data, but it is throwing an error. I can't find any example of this use case. It isn't used in any of the examples or in the tutorial. Is this a false positive error, and a bug?

```
The GraphQL query from C:/Users/jbolda/Documents/GitHub/jacobbolda.com/src/templates/cfBlogP
ost.js failed

        Errors:
          GraphQLError: Cannot read property 'id' of undefined
        Query:
          query cfBlogPostByID(
  $id: String!
  $heroImage: String!
) {
  post: contentfulBlogPost(id: {eq: $id}) {
    title
    content {
      childMarkdownRemark {
        html
      }
    }
    ...cfBlogPost
  }
  hero: file(relativePath: {eq: $heroImage}) {
    childImageSharp {
      sizes {
        ...GatsbyImageSharpSizes_tracedSVG
      }
    }
  }
}
```

**Comment by remoz** [comment]
I've also encountered this issue. On the build in develop, Gatsby sends a bunch of 'id' of undefined errors even if the queries themselves aren't wrong, and even if the query doesn't ask for an id.

One interesting thing: I've noticed that queries inside the layouts folder aren't affected by this error. Queries inside the pages folder **do** trigger errors. 🤔  I'll explore it and see what I can figure out.
